core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1649,6 +1649,50 @@ private static SslContext findDefaultSslContext(VirtualHost defaultVirtualHost,
         return null;
     }
 
+    ServerConfig buildServerConfig(ServerConfig existingConfig) {
+        final AnnotatedServiceExtensions extensions =
+                virtualHostTemplate.annotatedServiceExtensions();
+
+        assert extensions != null;
+
+        final VirtualHost defaultVirtualHost =
+                defaultVirtualHostBuilder.build(virtualHostTemplate);
+        final List<VirtualHost> virtualHosts =
+                virtualHostBuilders.stream()
+                        .map(vhb -> vhb.build(virtualHostTemplate))
+                        .collect(toImmutableList());
+        // Pre-populate the domain name mapping for later matching.
+        final Mapping<String, SslContext> sslContexts;
```

Comment:
Because we rebuild the `virtualHosts`, we should create the `sslContexts` again.
How about:
- Adding `sslContexts` into the `ServerConfig`
- Extracting the same logic from `buildServerConfig(ServerConfig existingConfig)` and `build()`?
```java
ServerConfig buildServerConfig(ServerConfig existingConfig) {
    return buildServerConfig(existingConfig.ports());
}

Server build() {
    return new Server(buildServerConfig(ports));
}

ServerConfig buildServerConfig(List<ServerPort> ports) {
    // Setting the parameters...

    return new ServerConfig(...);
}

```

Reply:
Good suggestion. I made the change.